src/qp/metrics/base_metric_classes.py

```diff
@@ -160,3 +161,44 @@ class PointToDistMetric(BaseMetric):
 
     def evaluate(self, estimate, reference):
         raise NotImplementedError()
+
+    def eval_from_iterator(self, estimate, reference):
+        self.initialize()
+        for estimate, reference in zip(estimate, reference):
+            self.accumulate(estimate, reference)
+        return self.finalize()
+
+    def initialize(self):
+        pass
+
+    def accumulate(self, estimate, reference):
+        raise NotImplementedError()
+
+    def finalize(self):
+        raise NotImplementedError()
+
+
+class PointToPointMetricDigester(PointToPointMetric):
+
+    def accumulate(self, estimate, reference):
+        # centroids = estimate.ancil['z_mode'] - reference
+        # digest = TDigest.make_centroied(centroids)  # Or something like this
+        digest = TDigest.compute(estimate, compression=100)
```

Comment:
So, I think that "estimate" is probably going to be a column in a Table, i.e., an array, and reference is too.   Probably we are going to want some function of both estimate and reference, as in the special case.  If it is always the same function, then it makes sense to have an intermediate class, but if it is a different function each time, then we don't get anything from having the intermediate class.

Reply:
Gotcha, yes, that makes sense to me.